cedar-policy-core/src/entities.rs

```diff
@@ -216,6 +226,109 @@ impl Entities {
         dot_str.write_str("}\n")?;
         Ok(dot_str)
     }
+
+    /// Attempt to eagerly compute the values of attributes for all entities in the slice.
+    /// This can fail if evaluation of the [`RestrictedExpr`] fails.
+    /// In a future major version, we will likely make this function automatically called via the constructor.
+    pub fn evaluate(self) -> std::result::Result<Self, EvaluationError> {
```

Comment:
It might be useful for this to return a different type than `Self` (at least a newtype wrapper), so that the type system actually distinguishes between "unevaluated Entities" and "evaluated Entities" objects.

Reply:
But we need `is_authorized to still accept both, no?

Reply:
Ideally I'd love to live in a world where only the evaluated Entities are exposed, and thus all is_authorized callers always pass the evaluated Entities (because they're automatically evaluated during construction of `Entities` object).  But I understand we can't live in that world until next major.  So yes, I take your point